src/thunderbots/software/ai/hl/stp/evaluation/ball.cpp

```diff
@@ -13,4 +13,16 @@ namespace Evaluation
     {
         return ball.position().x() >= field.centerPoint().x();
     }
+
+    bool ballInFriendlyCorner(const Field &field, const Ball &ball, double radius)
+    {
+        return ball.position().isClose(field.friendlyCornerPos(), radius) ||
+               ball.position().isClose(field.friendlyCornerNeg(), radius);
+    }
+
+    bool ballInEnemyCorner(const Field &field, const Ball &ball, double radius)
+    {
+        return ball.position().isClose(field.enemyCornerPos(), radius) ||
+               ball.position().isClose(field.enemyCornerNeg(), radius);
+    }
```

Comment:
Is it correct to say the ball is in the corner if it's off the field?

Reply:
for ball in friendly/enemy half we just compare relative to center point where the ball is, we don't check if its actually on the field so I just used the same idea for the definition of a "corner"

Should we check if its off the field?

Reply:
@MathewMacDougall

Reply:
The general plan was to use this for detecting "corner kick" type cases for Play selection and any other behavior we might want to be different when the ball or something is in the corner. I didn't consider the ball being outside the field.

Since the ball is out of play when it is outside the field, I don't think there's any cases where if we didn't report it as being "in the corner" it would cause problems. The closest case I can think of is when the referee places the ball for a kick. Sometimes they put it kinda close to the edge of the field, but it should always be inside.

Now that you mention it, I think my initial intuition if I read this function for the first time would be that the ball is only in the corner if it is also in the field.

So I'd say also check if the ball is in the field. If the ball is not in the field, it should be considered *not* in the corner.

Reply:
added